Show guidance when subtitles aren't available

robsquires opened the pull request:
Adds a guidance message to videos that don't have subtitles (aka captions)/
The message is either displayed on the placeholder or in a floating banner in the scenarios where the video is autoplayed.

![Screenshot 2019-07-05 at 12 54 53](https://user-images.githubusercontent.com/616321/60721290-5595d580-9f25-11e9-850c-c5bc82412c03.png)

![Screenshot 2019-07-05 at 13 29 00](https://user-images.githubusercontent.com/616321/60722472-f0dc7a00-9f28-11e9-8c73-8815745e904c.png)

robsquires committed:
add state to identify if video was started via autoplay

robsquires committed:
render guidance component on placeholder

robsquires committed:
style guidance banner

robsquires committed:
rejig things to handle video reloading on homepage

robsquires committed:
convert guidance to a class

robsquires committed:
style close button

robsquires committed:
WIP tests

robsquires committed:
hover state on close button

robsquires committed:
close banner after 5 seconds

robsquires committed:
wrap guidance in mixin

robsquires committed:
hide banner when it auto closes
this way screen readers can still access it

robsquires committed:
add tests for guidance

robsquires committed:
link stright to video transcripts section

robsquires committed:
tidying

robsquires committed:
linting

robsquires committed:
use font smoothing to match the placeholder

JakeChampion commented on bower.json:
```diff
@@ -12,6 +12,7 @@
     "o-viewport": "^3.0.1",
-    "o-normalise": "^1.6.2"
+    "o-normalise": "^1.6.2",
+    "o-buttons": "^5.16.6"
```
Adding a new dependency technically makes this a breaking change and requires the release to be a new major version. This is because a consumer of o-video could be using o-buttons already and we would like to avoid a conflict being returned to the consumer (if they were using the origami build service, it would cause their website to break if a conflict was served 😔 )

robsquires replied:
Great to know! I've managed to remove the need for o-buttons all together 👍

JakeChampion replied:
If that's the case, you need to remove `o-buttons` from `bower.json` as well, otherwise it is still a major version change.

JakeChampion commented on src/scss/_guidance.scss:
```diff
@@ -0,0 +1,77 @@
+        @include oTypographySans(-1);
+        background-color:  oColorsMix($color: 'black', $background: 'transparent', $percentage: 75);
+        padding: 10px 16px;
+        -webkit-font-smoothing: antialiased;
```
```suggestion
        // sass-lint:disable no-vendor-prefixes
        -webkit-font-smoothing: antialiased;
        // sass-lint:enable no-vendor-prefixes
```

robsquires replied:
🤦‍♂ Had this in a rebased commit I hadn't to force pushed

robsquires committed:
remove need for o-buttons

JakeChampion commented on src/js/video.js:
```diff
@@ -308,6 +312,8 @@ class Video {
 
 		this.videoEl.src = this.rendition && this.rendition.url;
+		this.guidance.removeBanner();
+		this.videoEl.addEventListener('playing', this.showGuidanceBanner.bind(this), { once: true });
```
`addEventListener`'s once option is not supported in IE11 and not supported by polyfill.io either. You will likely need another solution for IE11 to use.

robsquires committed:
clear timeout if banner removed early

- prevents banner disappearing early when a skipped to the next video
in a playlist before the previous banenr was hidden

robsquires committed:
implement listen once

- `once` option not suppported by IE 11
